src/parser/priest/discipline/modules/shadowlands/covenants/BoonOfTheAscended.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+
+import ItemHealingDone from 'interface/ItemHealingDone';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+
+
+import SPELLS from 'common/SPELLS';
+import { formatNumber } from 'common/format';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { HealEvent } from 'parser/core/Events';
+import { Options } from 'parser/core/Module';
+
+import AtonementDamageSource from '../../features/AtonementDamageSource';
+import isAtonement from '../../core/isAtonement';
+
+class BoonOfTheAscended extends Analyzer {
+  static dependencies = {
+    atonementDamageSource: AtonementDamageSource,
+  };
+
+  protected atonementDamageSource!: AtonementDamageSource;
+
+  atonementHealing = 0;
+  directHealing = 0;
+  damageDealt = 0; 
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasCovenant(COVENANTS.KYRIAN.id);
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
+  }
+
+  onHeal(event: HealEvent) {
+    if (isAtonement(event)) {
+
+      const atonenementDamageEvent = this.atonementDamageSource.event;
+
+      if (!atonenementDamageEvent || atonenementDamageEvent.ability.guid !== SPELLS.ASCENDED_BLAST.id 
+        || atonenementDamageEvent.ability.guid !== SPELLS.ASCENDED_NOVA.id 
+        || atonenementDamageEvent.ability.guid !== SPELLS.ASCENDED_ERUPTION.id) {
+        return;
+      }
+
+      this.atonementHealing += event.amount + (event.absorbed || 0);
+      return;
+    }
+
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.ASCENDED_BLAST.id) {
+      this.directHealing += event.amount + (event.absorbed || 0);
+      return;
+    }
+    
+    if (spellId === SPELLS.ASCENDED_NOVA.id) {
+      this.directHealing += event.amount + (event.absorbed || 0);
+      return;
+    }
+    if (spellId === SPELLS.ASCENDED_ERUPTION.id) {
+      this.directHealing += event.amount + (event.absorbed || 0);
+      return;
+    }
+  }
```

Comment:
the entire onHeal should be rewritten after the eventlistener has been updated to be 
```js
onHeal(event: HealEvent) {
  if (isAtonement(event)) {

      const atonenementDamageEvent = this.atonementDamageSource.event;

      if (!atonenementDamageEvent) {
        return;
      }
      this.atonementHealing += event.amount + (event.absorbed || 0);
  } else {
    this.directHealing += event.amount + (event.absorbed || 0);
}
```